app/src/app/serverActions/serverActionExample.ts

```diff
@@ -0,0 +1,28 @@
+"use server";
+
+interface FormDataState {
+  name: string;
+  email: string;
+}
+
+export async function updateServerData(
+  prevState: FormDataState,
+  formData: FormData
+): Promise<FormDataState> {
+  console.log("prevState => ", prevState);
+  console.log("formData => ", formData);
+
+  const name = formData.get("name") as string;
+  const email = formData.get("email") as string;
+
+  // In a real application, you would typically perform
+  // some server mutation.
+  await Promise.resolve();
```

Comment:
If we want to tie in with `useFormStatus` / `pending state - we can simulate a delay here instead of an immediate resolve